hy/models.py

```diff
@@ -121,22 +121,39 @@ def __init__(self, string):
 _wrappers[type(None)] = lambda foo: HySymbol("None")
 
 
-class HyKeyword(HyObject, str_type):
+class HyKeyword(HyObject):
     """Generic Hy Keyword object. It's either a ``str`` or a ``unicode``,
     depending on the Python version.
     """
 
-    PREFIX = "\uFDD0"
+    __slots__ = ['_value']
 
-    def __new__(cls, value):
-        if not value.startswith(cls.PREFIX):
-            value = cls.PREFIX + value
-
-        obj = str_type.__new__(cls, value)
-        return obj
+    def __init__(self, value):
+        if value[0] != ':':
+            value = ':' + value
```

Comment:
Shouldn't the empty keyword be built with `value` set to the empty string instead of `":"`?

Reply:
I did this as its the easier repr function. I think your right that it makes sense, so I'll work on it, but what would we prefer to see: `HyKeyword(':foo')` or `HyKeyword('foo')`?

Currently they're actually equivalent, but maybe I should put some extra effort into making sure there's only one way.

Reply:
The colon is mandatory, so I think `HyKeyword('foo')` should produce `:foo`, whereas `HyKeyword(':foo')` would produce `::foo`.

Reply:
:+1: was thinking the same but I think thats a deeper change it seems. Might as well fix it though.